controllers/profiles/getOtherProfilePublicByUsername.js

```diff
@@ -0,0 +1,69 @@
+const {User, sequelize} = require('../../databases/models');
+const {checkMoreOrLess} = require('../../helpers/checkMoreOrLess');
+
+module.exports = async (req, res) => {
+  try {
+    const user = await User.findOne({
+      where: {username: req.params.username},
+      attributes: {
+        exclude: [
+          'human_id',
+          'is_backdoor_user',
+          'encrypted',
+          'created_at',
+          'updated_at',
+          'real_name',
+          'last_active_at',
+          'status',
+          'is_banned'
+        ]
+      }
+    });
+    if (user === null) {
+      return res.status(404).json({
+        code: 404,
+        status: 'error',
+        message: 'User not found'
+      });
+    }
+
+    const targetUserId = user?.dataValues?.user_id;
+
+    const getFollowerCountQuery = `SELECT COUNT(user_follow_user.user_id_follower) as count_follower from user_follow_user WHERE user_id_followed = :user_id`;
+    const getFollowingCountQuery = `SELECT COUNT(user_follow_user.user_id_followed) as count_following from user_follow_user WHERE user_id_follower = :user_id`;
+
+    const getFollowerCount = await sequelize.query(getFollowerCountQuery, {
+      replacements: {user_id: targetUserId}
+    });
+
+    const getFollowingCount = await sequelize.query(getFollowingCountQuery, {
+      replacements: {user_id: targetUserId}
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

These two queries are almost identical and could be abstracted into a function to follow the DRY (Don't Repeat Yourself) principle. This would make the code more maintainable and less prone to errors.
```javascript
async function getCount(query, userId) {
  const result = await sequelize.query(query, {
    replacements: {user_id: userId}
  });
  return result?.[0]?.[0];
}

const getFollowerCountResult = await getCount(getFollowerCountQuery, targetUserId)?.count_follower;
const getFollowingCountResult = await getCount(getFollowingCountQuery, targetUserId)?.count_following;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->